src/ui/components.tsx

```diff
@@ -246,8 +246,7 @@ const SingleTag = (props: {
 
   const [tagOptions, setTagOptions] = React.useState<Array<SelectableValue<string>>>([]);
   const [groupOptions, setGroupOptions] = React.useState<Array<SelectableValue<string>>>([]);
-  // we just use this state for notifying useEffect of the changed data. TODO: perhaps there is a better way?
```

Comment:
nice! i'm still a bit skeptical: did you test this in the ui? is the behavior the same?

Reply:
I did test that the autocompleters are retriggered when you change the group. That was all this small portion was about, wasn't it?

Reply:
yes, should be fine then.